### PR TITLE
Support partial rebuilds (right click build project in VS) for nuprojs.

### PR DESCRIPTION
**Pull request by bbowman**
This means switching from a run after everything model to delete to timestamp file
to a run before to create it one.

### CODE REVIEW

File: common/NuprojInitializer.msbuildproj [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
-<Project ToolsVersion="15.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+<Project ToolsVersion="14.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
```
Comment (rajsesh-msft) [inline common/NuprojInitializer.msbuildproj]: >14 [](start = 23, length = 2)

is this by design?

Comment (memontic-ms) [inline common/NuprojInitializer.msbuildproj]: Please keep 15.0

---
In reply to: [107542980](https://github.com/Microsoft/WinObjC/pull/2300#discussion_r107542980) [](ancestors = 107542980)

Comment (bbowman) [inline common/NuprojInitializer.msbuildproj]: nope!

File: common/NuprojInitializer.msbuildproj [hunk 2]
```diff
@@ -38,17 +38,18 @@
...
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'" />
   <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
-  <Target Name="CoreCompile" />
   <Target Name="CreateManifestResourceNames" />
+  <Target Name="CoreCompile" />
   <Target Name="GetReferenceAssemblyPaths" />
   <Target Name="CopyFilesToOutputDirectory" />
   <Target Name="_GenerateCompileDependencyCache" />
   
   <!-- This is where the fun starts. Any custom generic msbuild project goop goes here. -->
```
Comment (memontic-ms) [inline common/NuprojInitializer.msbuildproj]: > [](start = 2, length = 89)

Is this comment still relevant?

**Comment by rajsesh-msft** [comment]
What is the behavior and I do a Build solution (Ctrl + Shift + B)? Will we end up re-packaging everything?

File: tools/legacy/ObjectiveC/Legacy_ObjectiveC.csproj [hunk 3]
```diff
@@ -25,6 +25,7 @@
     <TargetFrameworkVersion>v4.5</TargetFrameworkVersion>
     <FileAlignment>512</FileAlignment>
     <GeneratePkgDefFile>false</GeneratePkgDefFile>
+    <DeployExtension>False</DeployExtension>
```
Comment (memontic-ms) [inline tools/legacy/ObjectiveC/Legacy_ObjectiveC.csproj]: ><DeployExtension>False</DeployExtension> [](start = 4, length = 40)

Is this for build machine reasons?

Comment (bbowman) [inline tools/legacy/ObjectiveC/Legacy_ObjectiveC.csproj]: no I was seeing it locally and was upset. 

---
In reply to: [107544404](https://github.com/Microsoft/WinObjC/pull/2300#discussion_r107544404) [](ancestors = 107544404)

**Comment by rajsesh-msft** [comment]
Got the answer offline. Yes, we will create new nuget packages.

---
In reply to: [288553358](https://github.com/Microsoft/WinObjC/pull/2300#issuecomment-288553358) [](ancestors = 288553358)

**Comment by bbowman** [comment]
the same as before (it will re package things ). 

---
In reply to: [288553358](https://github.com/Microsoft/WinObjC/pull/2300#issuecomment-288553358) [](ancestors = 288553358)

**Pull request merged**